### ISSUE TITLE
`object-curly-newline` allow options common for both ObjectExpression and ObjectPattern to be defined as root level option

### ISSUE THREAD

**Issue by jrencz** [eslint/eslint#6770]
**What version of ESLint are you using?**
3.1.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
(Partial configuration for mentioned rule only)

``` js
"object-curly-newline": ["error", {
  "multiline": true,
  "ObjectExpression": {
    "minProperties": 2,
  },
  "ObjectPattern": {
    "minProperties": 5,
  },
}],
```

**What did you do? Please include the actual source code causing the issue.**
I enabled the `object-curly-newline` rule

**What did you expect to happen?**
Rule works, `multiline` option is applied to both `ObjectExpression` and `ObjectPattern`

**What actually happened? Please include the actual, raw output from ESLint.**
I got

```
 throw new Error(message.join(""));
        ^

Error: /Volumes/Projects/<path cropped>/.eslintrc.js:
    Configuration for rule "object-curly-newline" is invalid:
    Value "[object Object]" no (or more than one) schemas match.
```

It would be reasonable not to duplicate knowledge.

**Comment by mysticatea** [comment]
Thank you for this issue.

This is intentional behavior at this point.
I thought a mix of types (`ObjectExpression` and `ObjectPattern`) and other settings is confusing.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
